[Mack34021] iP

Mack34021 opened the pull request.

Mack34021 committed:
Rename Duke to Baraleous. Track Baraleous.java

Mack34021 committed:
Add echo function

Mack34021 committed:
Track TaskList.java. Implement basic task list

Can add to task list, print entire task list

Mack34021 committed:
Track Task.java. Implement mark/unmark-ing of tasks

Changed tasks from just strings to classes with a string and a boolean for marked status

Mack34021 committed:
Conform to coding standards

samika2005 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Baraleous {
```
creative name!

asytrix99 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+            String userInput = scanner.nextLine();
+            ArrayList<String> commandsList = getArguments(userInput);
+            switch (commandsList.get(0)){
+            case "bye":     // close program
```
Good comment indentation relative to position in the code

asytrix99 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+public class Baraleous {
+    public static void main(String[] args) {
+        String logo = "Hello! I'm Baraleous XIV!\n" +
+                "What can I do for you today?";
```
Should the `+` be on the next line? Refer to Layout under [https://se-education.org/guides/conventions/java/basic.html](url)

zeeeing replied:
Suggest defining this as a class constant: `private static final String LOGO = "..."` (move out of `main`, use `UPPER_SNAKE_CASE`, and reuse via `LOGO`)

asytrix99 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+                if (commandsList.size() == 2){
+                    String indexToMark = commandsList.get(1);
+                    markTaskDone(indexToMark, taskList);
+                }else if (commandsList.size() > 2){
```
Should there be a space in between `}` and else `?` Same scenario for other `else if`'s

asytrix99 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+                break;
+            default:        // Requested new task added
+                taskList.addTaskToList(userInput.trim());
+                printMessage("Added: " + userInput, false);
```
Similar case to lines 6-7, on Layout on wrapped lines indentation

asytrix99 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+            return;
+        }
+        if (taskIndex >= taskList.getTaskListLength()){
+            printMessage("Brother you are not that busy, you don't have " + taskIndex + " tasks, you only got " + (taskList.getTaskListLength()-1), false);
```
Similar case to lines 6-7, on Layout on wrapped lines indentation

samika2005 replied:
it only gets better! brilliant!

asytrix99 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+        printMessage(logo, false);
+        Scanner scanner = new Scanner(System.in);
+        TaskList taskList = new TaskList();
+        commandLoop: while(true){
```
Should you leave a space in between `while` and `(true)`

zeeeing replied:
Adding to the above comment, space in between `(true)` and `{` as well:

```suggestion
        commandLoop: while (true) {
```

asytrix99 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+     */
+    private static void markTaskDone(String taskToMark, TaskList taskList) {
+        int taskIndex = Integer.parseInt(taskToMark);
+        if (taskIndex <= 0){
```
Should you leave a space in between `)`and `{`?

asytrix99 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+        }
+        Task curTask = taskList.getTaskFromList(taskIndex);
+        if (!curTask.isTaskDone()){
+            printMessage("Hey! That's already not complete\n[ ] " + curTask.getTaskString(), false);
```
Similar case to lines 6-7, on Layout on wrapped lines indentation

asytrix99 reviewed (commented):
Overall, just some nits - line wrapping & spacings. Otherwise, LGTM!

zeeeing reviewed (commented):
I think the overall it's good, main issue would be the spacings in your codebase. May want to refer to [here](https://se-education.org/guides/conventions/java/basic.html) for more information.

Mack34021 committed:
Track Deadline.java, Event.java. Implement Deadlines and events

Deadline and Event extend Task with assoiated start/finish times

samika2005 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+                }else if (commandsList.size() > 2){
+                    printMessage("... Thats too many arguments", false);
+                }else{
+                    printMessage("... Which task do you want me to mark???? Not a mind reader....", false);
```
the mind reader bit is so unironically brilliant

samika2005 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+    private static void markTaskDone(String taskToMark, TaskList taskList) {
+        int taskIndex = Integer.parseInt(taskToMark);
+        if (taskIndex <= 0){
+            printMessage("Mate.... task " + taskIndex + " ..... really? Indexing starts at task 1.", false);
```
that's genius

samika2005 commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,160 @@
+
+    /**
+     * Prints to terminal in a nice formatted manner.
+     * @param message String to be printed to the user
```
really helpful and descriptive comments,gg!

samika2005 reviewed (commented):
gg overall!

Mack34021 committed:
Implement automated testing

Tests: todo, deadline, event, list, mark commands

Mack34021 committed:
Refactor code, remove deep nesting

As per pull request review, some deep nesting removed, as well as general coding standards conformation

Mack34021 committed:
Update Readme

Mack34021 committed:
Refactor variables

Move Logo text into constant variable

Mack34021 committed:
Merge branch 'master' into branch-Level-5

LuoYu-uwu commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,310 @@
+            addEventToTaskList(commandsList, taskList);
+            break;
+        case "task":    // Requested new task added
+            StringBuilder strBuil = new StringBuilder();
```
Perhaps consider giving a more descriptive name for `strBuil` that explains the variable better

LuoYu-uwu commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,310 @@
+     * @param userInput    The unsplit user input string.
+     * @return True if it needs to terminate early for program exit. False otherwise
+     */
+    private static boolean commandSwitch(ArrayList<String> commandsList, TaskList taskList, String userInput) {
```
Perhaps can consider renaming `commandSwitch` to sound more like boolean

LuoYu-uwu commented on src/main/java/Baraleous.java:
```diff
@@ -0,0 +1,310 @@
+
+        // Trim all strings, then create and add deadline to the task list
+        if (eventNameTrimmed.isEmpty()) {
+            eventNameTrimmed = "UNSPECIFIED";   // In case user forgets to give due date
```
Perhaps can consider using a constant for `"UNSPECIFIED"` to avoid magic literals since it is repeated a few times

LuoYu-uwu reviewed (commented):
Overall, code is clear and mostly follows Java standards and good coding practices. Good Job!

Mack34021 committed:
Check some inpu errors

Mack34021 committed:
Merge branch 'branch-Level-5'

Mack34021 committed:
Refactor into Packages

Mack34021 committed:
Merge branch 'branch-A-Packages'

Mack34021 committed:
Implement task deleting

Mack34021 committed:
Save to text file when updating task list

Mack34021 committed:
Read from save file on program startup

Mack34021 committed:
Add assets folder if not already there

Mack34021 committed:
Merge branch 'branch-Level-7'

Mack34021 committed:
Merge branch 'branch-Level-6'

Mack34021 committed:
Add save-file functionality to delete command

Mack34021 committed:
Jar file now working with level 6,7

Mack34021 committed:
Add more objects, Delete redundant files

Mack34021 committed:
Implement Search Function

Mack34021 committed:
Clean Level-9

Mack34021 committed:
Merge pull request #1 from Mack34021/Level-9

Clean Level-9

Mack34021 committed:
Add Javadoc

Mack34021 committed:
Merge pull request #2 from Mack34021/A-JavaDoc

Add Javadoc

Mack34021 committed:
Remove some unneccessary prints

Mack34021 committed:
Update Readme

Mack34021 committed:
Update Readme

Mack34021 committed:
Move readme to docs/

Mack34021 committed:
Fix bug in removeTaskFromList
